Database/Updates/World/06-06-27-2017-generator-chains-testdata.sql

```diff
@@ -0,0 +1,122 @@
+  CONSTRAINT `fk_generator_link__AceObject` FOREIGN KEY (`aceObjectId`) REFERENCES `ace_object` (`aceObjectId`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+/* Remove potential old stuff */
+
+DELETE FROM ace_weenie_class
+WHERE weenieClassId in (893, 954, 1996, 4172);
+
+DELETE FROM ace_object
+WHERE aceObjectId in (893, 954, 1996, 4172);
+
+/* Insert new stuff */
+INSERT INTO ace_weenie_class
+	(weenieClassId,
+	weenieClassDescription)
+VALUES 
+	(893, 'drudgeskulkergen'),
+	(954, 'drudgesneakergen'),
+	(1996, 'lowaaluviangen'),
+	(4172, 'drudgecampgen');
+	
+INSERT INTO ace_object
+	(aceObjectId,
+	aceObjectDescriptionFlags,
+	weenieClassId,
+	weenieHeaderFlags,
+	physicsDescriptionFlag)
+VALUES
+	(893, 148, 893, 0, 32769),
+	(954, 148, 893, 0, 32769),
+	(1996, 148, 893, 0, 32769),
+	(4172, 148, 893, 0, 32769);
+	
+INSERT INTO ace_object_properties_did
+	(aceObjectId,
+	didPropertyId,
+	propertyValue)
+VALUES 
+	(893, 1, 33555051),
+   (893, 8, 100667494),
+	(954, 1, 33555051),
+   (954, 8, 100667494),
+   (1996, 1, 33555051),
+   (1996, 8, 100667494),
+	(4172, 1, 33555051),
+   (4172, 8, 100667494);
+
+INSERT INTO ace_object_properties_int
+	(aceObjectId,
+	intPropertyId,
+	propertyValue)
+VALUES 
+	(893, 1, 0),      /* ItemType = 0 */
+	(893, 81, 1),     /* MaxGeneratedObjects = 1 */
+	(893, 100, 2),    /* GeneratorType = Absolute */
+	(893, 104, 7),		/* ActivationCreateClass = 7 (wcid of Drudge Skulker) */
+	(893, 142, 0),    /* GeneratorTimeType = Undef */
+	(893, 9006, 100), /* GeneratorProbability = 100 */
+	(954, 1, 0),      /* ItemType = 0 */
+	(954, 81, 1),     /* MaxGeneratedObjects = 1 */
+	(954, 100, 2),    /* GeneratorType = Absolute */
+	(954, 104, 940),  /* ActivationCreateClass = 940 (wcid of Drudge Sneaker) */
+	(954, 142, 0),    /* GeneratorTimeType = Undef */
+	(954, 9006, 100), /* GeneratorProbability = 100 */
+	(1996, 1, 0),      /* ItemType = 0 */
+	(1996, 81, 10),    /* MaxGeneratedObjects = 10 */
+	(1996, 100, 1),    /* GeneratorType = Relative */
+	(1996, 104, 0),  /* ActivationCreateClass = 0 (linked generators) */
+	(1996, 142, 0),    /* GeneratorTimeType = Undef */
+	(1996, 9006, 100), /* GeneratorProbability = 100 */
+	(4172, 1, 0),      /* ItemType = 0 */
+	(4172, 81, 3),     /* MaxGeneratedObjects = 3 */
+	(4172, 100, 2),    /* GeneratorType = Absolute */
+	(4172, 104, 0),	 /* ActivationCreateClass = 0 (linked generators) */
+	(4172, 142, 0),    /* GeneratorTimeType = Undef */
+	(4172, 9006, 100); /* GeneratorProbability = 100 */ 
+
+INSERT INTO ace_object_properties_string 
+	(aceObjectId,
+	strPropertyId,
+	propertyValue)
+VALUES 
+	(893, 1, 'Drudge Skulker Generator'),
+	(954, 1, 'Drudge Sneaker Generator'),
+	(1996, 1, 'Low A Aluvian Generator'),
+	(4172, 1, 'Drudge Camp Generator');
+	
+INSERT INTO ace_object_generator_link
+	(aceObjectId,
+	`index`,
+	generatorId,
+	generatorWeight)
+VALUES
+	(1996, 1, 4172, 100),
+	(4172, 1, 893, 60),
+	(4172, 2, 954, 40);
+	
+INSERT INTO ace_position 
```

Comment:
I can't think of any reason a weenie should have a posititontype=1 in based on our current setup. For every weenie we have, there's at least two objects in ace_object if the weenie was spawned in world, the weenie object and the instance object.

Following your lead, lowaaluviangen could be an example of a landscape generator that was place all over the outskirts of holtburg right? I'd think potentially 1 for each landblock. lets just say there are 5 landblocks surrounding holtburg, So thats the weenie object (1996) with no position, then 5 instances of that weenie with a position somewhere within the landblock, perhaps the center of it just to help with whatever location randomization code we go with. The instances would end up having generated aceObjectIds from the insert scripts to prevent collisions.

drudgecampgen could be nested with options to spawn campfires, drudgesneakergens and drudgeskulkergens, or even just the wcids of those drudges themselves instead of generators right? really sort of up to whomever is setting up the generator I guess and assumes the responsibility of not screwing up a spawn table that ripples across the world ;)

drudgecampgen could be a spawn option of the lowaaluviangen so technically it needs no positions although of course there would be instances of the drudgecampgen as well as drudgesneakergen and drudgeskulkergen because they are such basic generators that had to have been widely used in the game